tests/Service/MiraklClientTest.php

```diff
@@ -45,6 +45,28 @@ public function testGetNextLink()
         $link  = $this->miraklClient->getNextLink($responsePrevNext);
         $this->assertEquals($nextLink, $link);
     }
+
+    public function testGetMoreThanOneHundredOrdersById()
+    {
+        $orderIds = array_fill(0, 99, MiraklMockedHttpClient::ORDER_STATUS_WAITING_ACCEPTANCE);
+        $orderIds[] = MiraklMockedHttpClient::ORDER_STATUS_SHIPPING;
+        $orderIds[] = MiraklMockedHttpClient::ORDER_STATUS_SHIPPED;
+
+        $res = $this->miraklClient->listProductOrdersById($orderIds);
+
+        $this->assertEquals(array_keys($res), [MiraklMockedHttpClient::ORDER_STATUS_WAITING_ACCEPTANCE, MiraklMockedHttpClient::ORDER_STATUS_SHIPPING, MiraklMockedHttpClient::ORDER_STATUS_SHIPPED]);
```

Comment:
We don’t get that number of result objects out here because the duplicate order objects (duplicated in the tests because the mocked client only knows about a handful of order ids) get deduplicated when stored in a dictionary keyed by order id: https://github.com/stripe/stripe-mirakl-connector/blob/d61f1c2660f15da30e41acd6eadeecaa86801bd9/src/Service/MiraklClient.php#L135

In order to actually receive a large number of order objects here, the mock client could be modified to be able to dynamically generate order response objects given an arbitrary id, would that be the preferred approach?

Reply:
For listProductOrdersById, we can generate random IDs:
```
$orderIds = [];
for ($i = 0; $i <= 100; $orderIds[] = 'random_order_' . ++$i);
```

For listProductOrdersByCommercialId, the MiraklMock doesn't have a default commercial order if we generate random IDs so we'd have to also update this function: https://github.com/stripe/stripe-mirakl-connector/blob/master/tests/MiraklMockedHttpClient.php#L346
Though I don't think this is necessary if the listProductOrdersById test passes after the change above.

Reply:
Cool, I've made that modification to the `listProductOrdersById` test and left the `listProductOrdersByCommercialId` with spot-check assertions that the expected orders are present.